app/controllers/request_purpose.js

```diff
@@ -0,0 +1,58 @@
+import Ember from "ember";
+import AjaxPromise from 'browse/utils/ajax-promise';
+const { getOwner } = Ember;
+
+export default Ember.Controller.extend({
+  peopleCount: null,
+  description: "",
+  selectedId: null,
+  isSelfSelected: Ember.computed.equal("selectedId", "organisation"),
+  user: Ember.computed.alias('session.currentUser'),
+
+  actions: {
+    clearDescription() {
+      this.set("description", "");
+    },
+    
+    createOrderWithRequestPuropose(){
```

Comment:
Can we extract some functionality in another function?

Reply:
Yup Sure. Thanks for input. 👍